code/modules/security/genpop.dm

```diff
@@ -0,0 +1,721 @@
+			desc = initial(desc)
+			locked = FALSE
+			update_icon()
+			qdel(I)
+			playsound(src, 'sound/machines/terminal_insert_disc.ogg', 50, 0)
+		else
+			say("Access Denied. If you are a released prisoner, please insert your prisoner ID.")
+			return
+	else
+		return ..()
+
+/obj/structure/closet/secure_closet/genpop/AltClick(user)
+	var/mob/living/carbon/human/H = user
+	var/obj/item/card/id/I = H.get_idcard()
+	if(!registered_name)
+		if(allowed(user) && !istype(I, /obj/item/card/id/prisoner))
+			say("Invalid, please assign this locker to prisoners first before handling it!") // Prevents officers from "forgetting" to assign lockers to the prisoners they are handling.
+			return
+	..()
+
+
+/obj/machinery/turnstile/Initialize(mapload)
+	. = ..()
+	icon_state = "turnstile"
+	//Attach a signal handler to the turf below for when someone passes through.
+	//Signal automatically gets unattached and reattached when we're moved.
+	var/static/list/loc_connections = list(
+		COMSIG_ATOM_ENTERED = PROC_REF(on_entered),
+	)
+	AddElement(/datum/element/connect_loc, loc_connections)
+	AddComponent(/datum/component/simple_rotation, ROTATION_ALTCLICK | ROTATION_CLOCKWISE | ROTATION_COUNTERCLOCKWISE | ROTATION_VERBS, null, CALLBACK(src, PROC_REF(can_be_rotated)))
+
+/obj/machinery/turnstile/proc/can_be_rotated(mob/user, rotation_type)
+	if(!anchored && state == TURNSTILE_SHELL)
+		return TRUE
+	to_chat(user, "<span class='warning'>It is fastened to the floor!</span>")
+	return FALSE
+
+/obj/machinery/turnstile/proc/on_entered(datum/source, atom/movable/arrived, atom/old_loc, list/atom/old_locs)
+	SIGNAL_HANDLER
+	if(istype(arrived, /mob))
+		flick("operate", src)
+		playsound(src,'sound/items/ratchet.ogg',50,0,3)
+		if(arrived.pulledby) // don't do anything else for prisoners that are being escorted.
+			return
+		if(istype(arrived, /mob/living/carbon/human)) //we only want to prevent prisoners from being able to reuse their ID card.
+			var/mob/living/carbon/human/H = arrived
+			var/obj/item/card/id/id_card = H.get_idcard(hand_first = TRUE)
+			if(ACCESS_PRISONER in id_card?.GetAccess())
+				id_card.access -= list(ACCESS_PRISONER) //Prisoner IDs can only be used once to exit the turnstile
+				to_chat(H, "<span class='warning'>Your prisoner ID access has been purged, you won't be able to exit the prison through the turnstile again!</span>")
+				addtimer(CALLBACK(src, PROC_REF(exit_push), H), 2)
+
+/obj/machinery/turnstile/proc/exit_push(atom/movable/pushed) //Just "pushes" prisoners that are being released out of the turnstile so that they don't trap themselves.
+	var/movedir = turn(dir, 180) //Set to be the opposite of the turnstile dir, as that would always be the exit, unless the turnstile has been built incorrectly.
+	pushed.Move(get_step(pushed, movedir), movedir)
+
+///Handle movables (especially mobs) bumping into us.
+/obj/machinery/turnstile/Bumped(atom/movable/movable)
+	. = ..()
+	if(!istype(movable, /mob)) //just let non-mobs bump
```

Comment:
This comment does not match up with what it is doing. You are cancelling the proc if the object is not a mob, which means only mobs can use this proc.
Also consider using `ismob(movable)`